### PR TITLE
Remove sidebar

### PR DESCRIPTION
**Pull request by tomasmatus**
Rework of layout following mockup in #740

image

### CODE REVIEW

File: src/common.ts
Comment (cockpituous) [inline src/common.ts:66]: This added line is not executed by any test.

File: src/current-dir-detail.tsx
Comment (cockpituous) [inline src/current-dir-detail.tsx:22]: These 2 added lines are not executed by any test.

File: src/current-dir-detail.tsx
Comment (cockpituous) [inline src/current-dir-detail.tsx:47]: This added line is not executed by any test.

**Comment by jelly** [comment]
I assume this needs some follows up:

* [ ] allow clicking of permissions in the bottom bar or show a popover? (Or both)

**Review by jelly — COMMENTED** [review]
One issue found when I press `show hidden files` it still shows 4 hidden files. We did this different in the old sidebar.

image

Another issue I think is not showing a tooltip for the modification date of files, we should allow one to see the exact ISO formatted timestamp.

**Comment by tomasmatus** [comment]
> Another issue I think is not showing a tooltip for the modification date of files, we should allow one to see the exact ISO formatted timestamp.

Good idea

> allow clicking of permissions in the bottom bar or show a popover?

Yes, this was on my list to ask about as it wasn't mentioned in the mockup, makes sense to add it then

> One issue found when I press show hidden files it still shows 4 hidden files. We did this different in the old sidebar.

will do

File: src/common.ts
Comment (cockpituous) [inline src/common.ts:66]: This added line is not executed by any test.

File: src/current-dir-detail.tsx
Comment (cockpituous) [inline src/current-dir-detail.tsx:30]: These 2 added lines are not executed by any test.

**Comment by tomasmatus** [comment]
@garrett This should be ready for a design review. Could you please have a look? Tests are red because I am waiting for #775 after which I'lll fix last remaining test.

**Comment by garrett** [comment]
Looks great so far!

It doesn't seem to update when I select a file, however. (The sidebar did, and I thought the status bar would too. The default is to show the current directory if nothing is selected, but show the details for selected items when things are selected.)

File: src/files-footer-detail.tsx
Comment (cockpituous) [inline src/files-footer-detail.tsx:42]: These 2 added lines are not executed by any test.

**Comment by tomasmatus** [comment]
Hmm testingfarm red. Fifefox doesn't seem to like it without a sidebar :)

**Comment by jelly** [comment]
> Hmm testingfarm red. Fifefox doesn't seem to like it without a sidebar :)

```
Wrote HTML dump to TestFiles-testContextMenu-centos-10-stream-10.88.0.1-22-FAIL-2.html
Wrote JS log to TestFiles-testContextMenu-centos-10-stream-10.88.0.1-22-FAIL-2.js.log
Journal extracted to TestFiles-testContextMenu-centos-10-stream-10.88.0.1-22-FAIL-2.log.gz
> warn: transport closed: disconnected
umount: /run/user/1001: no mount point specified.
Exiting due to channel error.
Exiting due to channel error.
Exiting due to channel error.
Exiting due to channel error.
Exiting due to channel error.
Exiting due to channel error.
Traceback (most recent call last):
  File "/source/test/check-application", line 960, in testContextMenu
    b.click(".contextMenu button:contains('Create directory')")
  File "/source/test/common/testlib.py", line 575, in click
    self.mouse(selector + ":not([disabled]):not([aria-disabled=true])", "click")
  File "/source/test/common/testlib.py", line 506, in mouse
    self.wait_visible(selector)
  File "/source/test/common/testlib.py", line 814, in wait_visible
    self._wait_present(selector)
  File "/source/test/common/testlib.py", line 805, in _wait_present
    self.wait_js_func('ph_is_present', selector)
  File "/source/test/common/testlib.py", line 799, in wait_js_func
    self.wait_js_cond("%s(%s)" % (func, ','.join(map(jsquote, args))))
  File "/source/test/common/testlib.py", line 796, in wait_js_cond
    raise Error(f"timeout\nwait_js_cond({cond}): {last_error.msg}") from None
testlib.Error: timeout
wait_js_cond(ph_is_present(".contextMenu button:contains('Create directory'):not([disabled]):not([aria-disabled=true])")): Error: condition did not become true
```

The screenshot:

https://artifacts.dev.testing-farm.io/7bb50c34-603d-4432-bcea-f255b7917f44/work-allzupv3wj_/plans/all/execute/data/guest/default-0/test/browser-1/data/TestFiles-testContextMenu-centos-10-stream-10.88.0.1-22-FAIL-1.png

**Comment by garrett** [comment]
Looking at this again, hoping to unblock you, I noticed this:

image

When you hover over the file, it shows the URL we'll navigate to, which overlaps the summary. It's not ideal, but I don't know if there's anything we can do about this. I don't think we need to address it in this PR, and I'm not sure if we really can without breaking things.

At least in the single selection case, it's mainly redundant information in list view.

File: src/app.scss
Comment (garrett) [inline src/app.scss:59]: We could redo the grid layout with something simpler, using named tracks (starting and stopping lines on the endges to stretch to) instead of named spaces (the name of the content area), like this:

```css
.container {
  --gutter: 1rem;
  display: grid;
  grid-template: auto 1fr auto 0 / [full-start] var(--gutter) [content] 1fr var(--gutter) [full-end];
  gap: 0.5rem;
}

.toolbar {
  grid-column: full;
}

.content,
.footer {
  grid-column: content;
}
```

(PF spacers instead of rems, of course. And we could replace var(--gutter) with a PF spacer var too.)

The important things:

1. We don't need a separate layout for smaller widths, as desktop wide is the same layout as even the smallest mobile.
2. Horizontal is specified as full (all the way across, with a start and end) and content (only in the middle (not specifying start and end means it's just the 1 block instead).
3. This doesn't specify the vertical; that just falls into place based on in-document order.
4. The 0 is to use the gap at the end, similar to the `.` placeholder.

To make it more clear, we could also split the shorthand to :

```css
grid-template-columns: [full-start] var(--gutter) [content] 1fr var(--gutter) [full-end];
grid-template-rows: auto 1fr auto 0;
```

```css
grid-template-columns: [full-start] var(--gutter) [content-start] 1fr [content-end] var(--gutter) [full-end];
```

So, the above changes (exact same CSS layout, but perhaps more understandable) would then be:

```css
.container {
  --gutter: 1rem;
  display: grid;
  grid-template-rows: auto 1fr auto 0;
  grid-template-columns: [full-start] var(--gutter) [content-start] 1fr [content-end] var(--gutter) [full-end];
  gap: 0.5rem;
}

.toolbar {
  grid-column: full;
}

.content,
.footer {
  grid-column: content;
}
```

File: src/files-footer-detail.tsx
Comment (cockpituous) [inline src/files-footer-detail.tsx:42]: These 2 added lines are not executed by any test.

Comment (cockpituous) [inline src/files-footer-detail.tsx:42]: These 2 added lines are not executed by any test.

Comment (cockpituous) [inline src/files-footer-detail.tsx:42]: These 2 added lines are not executed by any test.

Comment (cockpituous) [inline src/files-footer-detail.tsx:42]: These 2 added lines are not executed by any test.

Comment (cockpituous) [inline src/files-footer-detail.tsx:42]: These 2 added lines are not executed by any test.

Comment (cockpituous) [inline src/files-footer-detail.tsx:42]: These 2 added lines are not executed by any test.

Comment (cockpituous) [inline src/files-footer-detail.tsx:42]: These 2 added lines are not executed by any test.

Comment (cockpituous) [inline src/files-footer-detail.tsx:42]: These 2 added lines are not executed by any test.

**Review by garrett — CHANGES_REQUESTED** [review]
The layout broke in the latest versions. If you have more content, it no longer scrolls properly; you have to scroll to the info bar.

File: src/app.scss
Comment (garrett) [inline src/app.scss:39]: This is the fix.

File: src/app.scss
Comment (garrett) [inline src/app.scss]: I consolidated all the `grid-column: content` directives into one multi-element selector.

File: src/app.scss
Comment (garrett) [inline src/app.scss]: As mentioned above, I moved this into a group of all the ones that use the "content" column.

**Review by garrett — APPROVED** [review]
OK, I've fixed the scroll issue, which was the last thing where I saw a problem, so I think it's probably good now. Please double-check. And, of course, squash things together where appropriate before merging. Thanks!

File: src/files-footer-detail.tsx
Comment (cockpituous) [inline src/files-footer-detail.tsx:42]: These 2 added lines are not executed by any test.

Comment (cockpituous) [inline src/files-footer-detail.tsx:42]: These 2 added lines are not executed by any test.

**Review by jelly — APPROVED** [review]

File: src/files-footer-detail.tsx
Comment (cockpituous) [inline src/files-footer-detail.tsx:42]: These 2 added lines are not executed by any test.

**Pull request merged**
